packages/story-editor/src/app/media/utils/useMediaInfo.js

```diff
@@ -0,0 +1,249 @@
+     * @return {Promise<MediaInfo|null>} File info or null on error.
+     */
+    async (file) => {
+      const getSize = () => file.size;
+
+      // TODO: Look into using createFileReader from media package.
+      const readChunk = (chunkSize, offset) =>
+        new Promise((resolve, reject) => {
+          const reader = new FileReader();
+          reader.onerror = reject;
+          reader.onload = (event) => {
+            if (event.target.error) {
+              reject(event.target.error);
+            }
+            resolve(new Uint8Array(event.target.result));
+          };
+          reader.readAsArrayBuffer(file.slice(offset, offset + chunkSize));
+        });
+
+      //eslint-disable-next-line @wordpress/no-unused-vars-before-return -- False positive because of the finally().
+      const trackTiming = getTimeTracker('load_mediainfo');
+
+      try {
+        // This will expose the window.MediaInfo global.
+        await loadScriptOnce(mediainfoUrl);
+
+        const mediaInfo = await window.MediaInfo({ format: 'JSON' });
+        const result = JSON.parse(
+          await mediaInfo.analyzeData(getSize, readChunk)
+        );
+
+        const normalizedResult = result.media.track.reduce(
+          (acc, track) => {
+            if (track['@type'] === 'General') {
+              acc.fileSize = Number(track.FileSize);
+              acc.format = track.Format.toLowerCase().replace('mpeg-4', 'mp4');
+              acc.frameRate = Number(Number(track.FrameRate).toFixed(0));
+              acc.codec = track.CodecID?.trim();
+            }
+
+            if (track['@type'] === 'Image' || track['@type'] === 'Video') {
+              acc.width = Number(track.Width);
+              acc.height = Number(track.Height);
+              acc.colorSpace = track.ColorSpace; // Maybe useful in the future.
+            }
+
+            if (track['@type'] === 'Video') {
+              acc.duration = Number(track.Duration);
+              acc.videoCodec = track.Format.toLowerCase();
+            }
+
+            if (track['@type'] === 'Audio') {
+              acc.audioCodec = track.Format.toLowerCase();
+            }
+
+            return acc;
+          },
+          {
+            mimeType: file.type,
+          }
+        );
+
+        normalizedResult.isMuted = !normalizedResult.audioCodec;
+
+        mediaInfo.close();
+
+        return normalizedResult;
+      } catch (err) {
+        // eslint-disable-next-line no-console -- We want to surface this error.
+        console.error(err);
+
+        trackError('mediainfo', err.message);
+
+        return null;
+      } finally {
+        trackTiming();
+      }
+    },
+    [mediainfoUrl]
+  );
+
+  const isConsideredOptimized = useCallback(
+    /**
+     * Determines whether a video file is to be considered optimized.
+     *
+     * Checks things like dimensions, file size, mime type, and codecs.
+     *
+     * @todo Allow WebM with VP9 once Safari catches up.
```

Comment:
Are you sure that it doesn't already support it - https://caniuse.com/webm

Reply:
Yes.

Check the footnotes. Safari on iOS currently only supports it for WebRTC. That's why the cells there aren't green.